ksql-engine/src/main/java/io/confluent/ksql/function/InternalFunctionRegistry.java

```diff
@@ -197,11 +197,26 @@ public synchronized AggregateFunctionFactory getAggregateFactory(final String fu
     return udafFactory;
   }
 
+  @Override
+  public TableFunctionFactory getTableFunctionFactory(final String functionName) {
```

Comment:
opposite comment 😂 should this be synchronized? otherwise we risk concurrent access on the `udtfs`

Reply:
+1